homeassistant/components/uptimerobot/config_flow.py

```diff
@@ -58,28 +58,54 @@ async def _validate_input(
             if response and response.data and response.data.email
             else None
         )
-        if account:
-            await self.async_set_unique_id(str(account.user_id))
-            self._abort_if_unique_id_configured()
 
         return errors, account
 
     async def async_step_user(self, user_input: ConfigType | None = None) -> FlowResult:
         """Handle the initial step."""
-        errors: dict[str, str] = {}
         if user_input is None:
             return self.async_show_form(
                 step_id="user", data_schema=STEP_USER_DATA_SCHEMA
             )
 
         errors, account = await self._validate_input(user_input)
         if account:
+            await self.async_set_unique_id(str(account.user_id))
+            self._abort_if_unique_id_configured()
             return self.async_create_entry(title=account.email, data=user_input)
 
         return self.async_show_form(
             step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
         )
 
+    async def async_step_reauth(
+        self, user_input: ConfigType | None = None
+    ) -> FlowResult:
+        """Rerun the user step when reauth is requested."""
+        return await self.async_step_reauth_confirm()
```

Comment:
Why extract a confirm function and not put the whole body in here ?

Reply:
Just following the docs

Reply:
Tests seem to lock up if this is not done, which seems like a bug somewhere.

Reply:
If if we don't have a separate step for the confirmation the initialization of the flow would pass the existing entry data to the form, which wouldn't be correct.